### PR TITLE
My Account Page

### PR DESCRIPTION
**Pull request by giteshk11**
# Issue Being Addressed
#83 

# Type of PR

Put an `x` in the brackets for the type(s) of PR this is. You may tick more than one.

[ ] Bug Fix
[ ] Refactor
[x] New Feature
[ ] Update to Existing Feature
[ ] Other (state below)

# Description

**For New Features:** What feature are we adding? Explain your technical approach to doing this.
Added My Account page for tenants so that tenants access there data and edit it

# How to Test/Reproduce

Provide steps on how one can test your changes here. e.g.
1. Log in with your account
2. Go to the account from bottom
3. Click on Edit profile to see the edit profile page

# Screenshots/casts

If applicable, include any relevant screenshot or screencasts of your changes.

Screen Shot 2020-08-31 at 14 37 28
Screen Shot 2020-08-31 at 12 09 43
Screen Shot 2020-08-31 at 14 37 33

### CODE REVIEW

**Comment by mort3za** [comment]
I don't see "Edit Profile"  

image

**Comment by paulchrisluke** [comment]
We hid it pending API support

File: src/components/tenant/EditProfileRow.vue [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<template>
```
Comment (mort3za) [inline src/components/tenant/EditProfileRow.vue:1]: *Optional* improvement: you can use `functional` component here. https://medium.com/js-dojo/vue-js-functional-components-what-why-and-when-439cfaa08713

File: src/layouts/WithNameBarLayout.vue [hunk 2]
```diff
@@ -0,0 +1,133 @@
+<template>
+  <div class="page min-h-screen flex flex-col text-white text-opacity-medium">
```
Comment (mort3za) [inline src/layouts/WithNameBarLayout.vue:2]: We don't have text-white in Figma, it probably should be `text-on-background`.

File: src/layouts/WithTitleBarLayout.vue [hunk 3]
```diff
@@ -4,7 +4,14 @@
...
       <!-- icon -->
-      <h1 class="tg-h2-mobile text-opacity-high flex-grow text-white ml-2">
+      <a
+        class="cursor-pointer"
+        @click.prevent="iconClick"
+        v-if="$route.meta.backRoute"
+      >
+        <IconBack class="text-white" />
+      </a>
+      <h1 class="tg-h2-mobile text-opacity-high flex-grow text-white">
```
Comment (mort3za) [inline src/layouts/WithTitleBarLayout.vue:14]: `text-white`: same

File: src/router/TenantPanelRoutes.js [hunk 4]
```diff
@@ -15,15 +13,23 @@ export const TenantPanelRoutes = [
     meta: { isPublic: true }
   },
   {
-    // tenant pmu setup
-    path: '/tenant/:tenantSlug/pmu-setup',
-    name: 'TenantPmuSetup',
-    component: () => import('@/views/TenantPmuSetup.vue'),
```
Comment (mort3za) [inline src/router/TenantPanelRoutes.js:21]: Why this got removed? Is this a mistake?

File: src/store/modules/tenant.js [hunk 5]
```diff
@@ -156,6 +157,31 @@ const actions = {
         }
       );
     });
+  },
+  userOwnsTenant(context, tenantSlug) {
+    let companySlug = process.env.VUE_APP_COMPANY_SLUG;
+    return new Promise((resolve, reject) => {
+      ajax.get(`/companies/${companySlug}/tenants/owns/${tenantSlug}`).then(
```
Comment (mort3za) [inline src/store/modules/tenant.js:164]: Use meredith-axios please (like the below action: BrowtricksTenantService.tenants), tell me if you had problem with it.

File: src/store/modules/tenant.js [hunk 6]
```diff
@@ -156,6 +157,31 @@ const actions = {
...
+    });
+  },
+  fetchUserTenant(context, tenantSlug) {
+    return new Promise((resolve, reject) => {
+      BrowtricksTenantService.tenants({ tenantSlug: tenantSlug }).then(
+        response => {
+          resolve(response);
+        },
+        error => {
+          reject(error);
```
Comment (mort3za) [inline src/store/modules/tenant.js:181]: We don't handle errors here, just return BrowtricksTenantService.tenants(... without defining a promise.

File: src/views/MyAccount.vue [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+  },
+  methods: {
+    ...mapActions('tenant', ['fetchUserTenant']),
+    init() {
+      this._fetchTenantData();
+    },
+    async _fetchTenantData() {
+      this.tenantData = await this.fetchUserTenant(
+        this.$route.params.tenantSlug
+      ).catch(() => {
```
Comment (mort3za) [inline src/views/MyAccount.vue:91]: We use try/catch block for catching errors with async/await. I didn't try this before, does it prevent "Unhandled promise rejection"?

**Review by mort3za — CHANGES_REQUESTED** [review]
Woo that's a nice PR! Thanks a lot, I enjoyed the way you adapted to the project.  
Added some comments

File: src/views/MyAccount.vue (hunk 7 above)
Comment (giteshk11) [reply src/views/MyAccount.vue:91]: I saw this somewhere in this code so I used similarly like that

**Review by paulchrisluke — CHANGES_REQUESTED** [review]
1. I seem unable to get back to my account after using the bottom nav for a bit

image

2. can we apply the same spinner transition to this page so it feels smooth?

3. hide the avatar section until the api is ready (just show the name)

**Pull request merged**